circuit_knitting/cutting/cut_finding/cco_utils.py

```diff
@@ -138,23 +138,20 @@ def greedy_best_first_search(
         Callable, search_space_funcs.next_state_func
     )
 
-    if search_space_funcs.goal_state_func(state, *args):
-        return state
-
-    best = min(
-        [
-            (search_space_funcs.cost_func(next_state, *args), k, next_state)
-            for k, next_state in enumerate(
-                search_space_funcs.next_state_func(state, *args)
-            )
-        ],
-        default=(None, None, None),
-    )
-
-    if best[-1] is None:  # pragma: no cover
-        # This covers a rare edge case.
-        # We have so far found no circuit that triggers it.
-        # Excluding from test coverage for now.
-        return None
-
-    return greedy_best_first_search(best[-1], search_space_funcs, *args)
+    while not search_space_funcs.goal_state_func(state, *args):
+        best = min(
+            [
+                (search_space_funcs.cost_func(next_state, *args), k, next_state)
+                for k, next_state in enumerate(
+                    search_space_funcs.next_state_func(state, *args)
+                )
+            ],
+            default=(None, None, None),
+        )
+        if best[-1] is None:  # pragma: no cover
+            # This covers a rare edge case.
+            # We have so far found no circuit that triggers it.
+            # Excluding from test coverage for now.
+            return None
```

Comment:
If you take my below comment you can just say:

```
state_out = None
break
```
and you won't have any returns in the loop anymore. This is another style choice, but as loops grow bigger and more return statements are added, it can become messy, so I've made a habit of pushing returns out of loops and setting appropriate variables within the loop.

Readability/understandability is the primary goal, so whatever you think is best here as well

Reply:
I actually tend to prefer a `return` in the loop over a `break` statement, so Ibrahim, it is truly up to you; as Caleb says, readability/understandability is the primary goal.